### PR TITLE
27418 mobile disability rating endpoint

### PR DESCRIPTION
**Pull request by jperk51**
## Description of change
Add disability endpoint to mobile module

## Original issue(s)
department-of-veterans-affairs/va.gov-team#27418

## Things to know about this PR
Added new spec and added endpoint to docs

### CODE REVIEW

**Review by kreek — APPROVED** [review]
Please add a spec that confirms the expected value of the endpoints response body, otherwise LGTM 🚀

File: modules/mobile/app/controllers/mobile/v0/disability_rating_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+require_dependency 'mobile/application_controller'
+
+module Mobile
+  module V0
+    class DisabilityRatingController < ApplicationController
+      def index
+        response = VeteranVerification::DisabilityRating.for_user(@current_user)
```
[the remark this answers is not in the document]
Comment (jperk51) [reply modules/mobile/app/controllers/mobile/v0/disability_rating_controller.rb:9]: I could rebuild it in our module if that would be better?

**Comment by kreek** [comment]
Now that we have the disability rating model in our module I think there's an opportunity to make some enhancements and align it with our patterns. Some of our, admittedly undocumented, rules (I should add this to the readme):

1.  Models (i.e. resources, non database backed) are immutable data buckets and should not fetch or adapt their own data.
2. Fetch via a proxy when reusing a service so we can make calls in parallel and customize error handling.
3. Perform data transformations in adapters.
4. Return the minimum amount of data.

So for this implementation:

1. Use `Common::Resource` as the model class to inherit from (`Virtus` is [no longer maintained](https://github.com/solnic/virtus#discontinued), `ActiveModel::Serialization` doesn't do anything since we use `FastJsonapi::ObjectSerializer`)
2. Proxy the `BGS::Services` class
3. Adapt the data (might be overkill for an adapter class a method could be fine) in that proxy and return a model
4. I believe we just need the combined disability rating (can confirm in the meeting today), only return that and the id :)

File: modules/mobile/app/models/mobile/v0/adapters/rating.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          elsif response[:disability_rating_record][:ratings].instance_of?(NilClass)
+            response[:disability_rating_record][:ratings] = []
+          end
+          ratings = response[:disability_rating_record][:ratings].map do |rating|
+            Mobile::V0::IndividualRating.new(
+              decision: rating[:disability_decision_type_name],
+              effective_date: get_formatted_date(rating[:begin_date]),
+              rating_percentage: rating[:diagnostic_percent].to_i,
+              diagnostic_text: rating[:diagnostic_text],
+              type: rating[:diagnostic_type_name]
```
Comment (jperk51) [inline modules/mobile/app/models/mobile/v0/adapters/rating.rb:36]: Currently returning both diagnostic_text and diagnostic_type_name. Given the cassette in vets-api and the logic/tests on web it's hard to determine which of these are the more appropriate choice. Once this stuff goes in I'll check a few staging users and patch this to only return the one we want to use.

File: modules/mobile/app/controllers/mobile/v0/disability_rating_controller.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    class DisabilityRatingController < ApplicationController
+      def index
+        response = rating_proxy.for_user
+        render json: Mobile::V0::DisabilityRatingSerializer.new(response)
+      end
+
+      private
+
+      def rating_proxy
+        @rating_proxy ||= Mobile::V0::DisabilityRating::Proxy.new(@current_user)
```
Comment (kreek) [inline modules/mobile/app/controllers/mobile/v0/disability_rating_controller.rb:16]: In Rails a new controller instance is created on each request. This means as long as a class' method isn't accessed twice we shouldn't need to memoize (@thing ||= ) it. Places where memoization helps would be when we make parallel calls.

File: modules/mobile/app/services/mobile/v0/disability_rating/proxy.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+module Mobile
+  module V0
+    module DisabilityRating
+      class Proxy
+        def initialize(user)
+          @user = user
+        end
+
+        def for_user
```
Comment (kreek) [inline modules/mobile/app/services/mobile/v0/disability_rating/proxy.rb:13]: `for_user` is a naming convention left over from when we returned an instance of a class with a local user reference. Could this describe the action taken? Maybe something like `.get_disability_ratings` or `.fetch_disability_ratings`.

**Review by kreek — APPROVED** [review]
Two minor comments but LGTM 🚀

**Review by kreek — APPROVED** [review]
LGTM 🚀

File: modules/veteran_verification/app/models/veteran_verification/disability_rating.rb [hunk 5]
```diff
@@ -30,7 +30,7 @@ def self.for_user(user)
     end
 
     def self.handle_errors!(response)
-      raise_error! unless response[:disability_rating_record].class.eql? Hash
+      raise_error! unless response[:disability_rating_record].instance_of?(Hash)
```
Comment (jperk51) [inline modules/veteran_verification/app/models/veteran_verification/disability_rating.rb:33]: Tried reverting these, but the linter kept being upset about them

**Review by Teesh — APPROVED** [review]

**Review by thilton-oddball — APPROVED** [review]

**Pull request merged**